project-base/storefront/components/Pages/Customer/deliveryAddressFormMeta.ts

```diff
@@ -0,0 +1,122 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+    validateCity,
+    validateCountry,
+    validateFirstName,
+    validateLastName,
+    validatePostcode,
+    validateStreet,
+    validateTelephoneRequired,
+} from 'components/Forms/validationRules';
+import useTranslation from 'next-translate/useTranslation';
+import { useMemo } from 'react';
+import { FieldError, UseFormReturn } from 'react-hook-form';
+import { DeliveryAddressFormType } from 'types/form';
+import { useShopsysForm } from 'utils/forms/useShopsysForm';
+import * as Yup from 'yup';
+
+export const useDeliveryAddressForm = (
+    defaultValues: DeliveryAddressFormType,
+): [UseFormReturn<DeliveryAddressFormType>, DeliveryAddressFormType | undefined] => {
+    const { t } = useTranslation();
+
+    const resolver = yupResolver(
+        Yup.object().shape<Record<keyof DeliveryAddressFormType, any>>({
+            firstName: validateFirstName(t),
+            lastName: validateLastName(t),
+            companyName: Yup.string(),
+            telephone: validateTelephoneRequired(t),
+            street: validateStreet(t),
+            city: validateCity(t),
+            postcode: validatePostcode(t),
+            country: validateCountry(t),
+        }),
+    );
+
+    return [useShopsysForm(resolver, defaultValues), defaultValues];
+};
+
+type DeliveryAddressFormMetaType = {
+    formName: string;
+    messages: {
+        error: string;
+        success: string;
+    };
+    fields: {
+        [key in keyof DeliveryAddressFormType]: {
+            name: key;
+            label: string;
+            errorMessage?: string;
+        };
+    };
+};
+
+export const useDeliveryAddressFormMeta = (
+    formProviderMethods: UseFormReturn<DeliveryAddressFormType>,
+): DeliveryAddressFormMetaType => {
+    const { t } = useTranslation();
+    const errors = formProviderMethods.formState.errors;
+
+    const formMeta = useMemo(
+        () => ({
+            formName: 'delivery-address-form',
+            messages: {
+                error: t('Error occured while saving your profile'),
```

Comment:
I would say `An error occurred...` (both the missing `An` and the missing double `rr`)